pkg/storagepolicy/sdkstoragepolicy.go

```diff
@@ -238,106 +262,163 @@ func (p *SdkPolicyManager) Enumerate(
 
 	policies := make([]*api.SdkStoragePolicy, 0)
 	for _, policy := range kvp {
-		volSpecs := &api.VolumeSpecPolicy{}
-		err = jsonpb.UnmarshalString(string(policy.Value), volSpecs)
+		sdkPolicy := &api.SdkStoragePolicy{}
+		err = jsonpb.UnmarshalString(string(policy.Value), sdkPolicy)
 		if err != nil {
 			return nil, status.Errorf(codes.Internal, "Json Unmarshal failed for policy %s: %v", policy.Key, err)
 		}
-		storagePolicy := &api.SdkStoragePolicy{
-			Name:   strings.TrimPrefix(policy.Key, policyPrefix+policyPath+"/"),
-			Policy: volSpecs,
+		// only enum volumes, owner has read access to
+		if sdkPolicy.IsPermitted(ctx, api.Ownership_Read) {
+			policies = append(policies, sdkPolicy)
 		}
-		policies = append(policies, storagePolicy)
 	}
 
 	return &api.SdkOpenStoragePolicyEnumerateResponse{
 		StoragePolicies: policies,
 	}, nil
 }
 
-// Enforce given storage policy
-func (p *SdkPolicyManager) Enforce(
+// SetDefault storage policy
+func (p *SdkPolicyManager) SetDefault(
 	ctx context.Context,
-	req *api.SdkOpenStoragePolicyEnforceRequest,
-) (*api.SdkOpenStoragePolicyEnforceResponse, error) {
+	req *api.SdkOpenStoragePolicySetDefaultRequest,
+) (*api.SdkOpenStoragePolicySetDefaultResponse, error) {
 	if req.GetName() == "" {
 		return nil, status.Error(codes.InvalidArgument, "Must supply a Storage Policy Name")
 	}
 
-	// verify policy exists, before enforcing
+	// verify policy exists, before setting it as default
 	policy, err := p.Inspect(ctx,
 		&api.SdkOpenStoragePolicyInspectRequest{
 			Name: req.GetName(),
 		},
 	)
 	if err != nil {
-		return nil, status.Errorf(codes.NotFound, "Policy with name %s not found", req.GetName())
+		return nil, err
+	}
+
+	// only administrator or owner can set policy as default
```

Comment:
Now that I am thinking about it, this affects the entire cluster, not just a resource. Affecting the cluster requires system admin ownership o.IsAdminByUser(). I think we meant that only the admin of the system (not a user with admin rights for a resource) can do this. If not, then anyone (any owner of a storage policy) with the correct RBAC can affect the cluster. What do you think?

Reply:
I guess that's right, setting/releasing default policy should be done by Admin user only. Anyone with write/Admin access can pass their storage policy for volume create/update